action.yaml

```diff
@@ -47,154 +49,222 @@ runs:
+
             if [[ -f $csv_file ]]; then
+              echo "---Producing Output Path"
+              # Creating the out path to store outputs.
+              if [[ $root_file == true ]]; then
+                out_path="out/${file_name}/"
+              else
+                out_path="out/${file_path}/${file_name}/"
+              fi
+              echo "out_path: ${out_path}"
+
+              echo "---Building CSV-W"
+              echo "Building CSV-W"
               if [[ -f $config_file ]]; then
                 echo "Config for ${csv_file} is available: ${config_file}"
-                echo "Building CSV-W"
-                csvcubed build $csv_file -c $config_file --out $out_path --validation-errors-to-file
+                csvcubed build "$csv_file" -c "$config_file" --out "$out_path" --validation-errors-to-file
               else
                 echo "Config for ${csv_file} is not available"
-                echo "Building CSV-W"
-                csvcubed build $csv_file --out $out_path --validation-errors-to-file
+                csvcubed build "$csv_file" --out "$out_path" --validation-errors-to-file
               fi
               
-              echo "Inspecting CSV-W"
+              echo "---Inspecting CSV-W"
               mapfile -d $'\0' inspectable_files < <(find "${GITHUB_WORKSPACE}/${out_path}" -name "*.csv-metadata.json" -type f -print0)
-              
-              for file in "${inspectable_files[@]}"; do
-                echo "Inspecting file ${file}"
-                file_path="${file%.*}"
-                file_name="${file_path##*/}"
-                inspect_output_file="${out_path}${file_name}_inspect_output.txt"
+              for inspect_file in "${inspectable_files[@]}"; do
+                echo "Inspecting file: ${inspect_file}"
+                inspect_file_path="${inspect_file%/*}"
+                inspect_file_name="${inspect_file##*/}"                  
+                inspect_output_file="${out_path}${inspect_file_name}_inspect_output.txt"
 
-                csvcubed inspect $file > $inspect_output_file
+                csvcubed inspect "$inspect_file" > "$inspect_output_file"
               done
               
-              echo "Finished processing file ${file}"
+              processed_files+=($csv_file)
+              processed_files+=($config_file)
+              echo "processed_files: ${processed_files[@]}"
+              
               echo "::set-output name=has_outputs::${{ toJSON(true) }}"
+              
+              echo "---Finished Processing File: ${file}"
+              echo "======================"                
             fi
           fi
         done
       shell: bash
 
-    - name: Process all deleted files
+    - name: Handle deleted files
       id: process-deleted-files
       run: |
         echo "::set-output name=has_outputs::${{ toJSON(false) }}"
         mapfile -d ',' -t deleted_files < <(printf '%s,' '${{ steps.get-added-changed-removed-files.outputs.removed }}')
```

Comment:
Should we be considering renamed files here as well? If they've been renamed, we will need to delete the old files won't we?

Reply:
Please see the comment below.